### ISSUE TITLE
convert big PDF documents slower in portable version 7.0.10.16 (non HDRI) than in v6.7.4 and 7.10.0.16

### ISSUE THREAD

**Issue by VassiliYakovlev** [ImageMagick/ImageMagick#2461]
### Prerequisites
[bigPDF.pdf](https://github.com/ImageMagick/ImageMagick/files/5117749/bigPDF.pdf)

- [x] I have written a descriptive issue title
- [x] I have verified that I am using the latest version of ImageMagick
- [x] I have searched [open](https://github.com/ImageMagick/ImageMagick/issues) and [closed](https://github.com/ImageMagick/ImageMagick/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

ImageMagick version 6.7.4 takes 16 sec to convert, version 7.0.10.16-portable - 42 sec, installed version 7.0.10.16 runs 19 sec, on my hardware, it can take less or more depends on hardware, but the patron is the same. New version runs almost tree times slower.

### Steps to Reproduce

Use attached PDF document as input file.
Execute this command on portable ImageMagick:
`convert  -density 150  -colorspace RGB  "bigPDF.pdf[0]"   -colorspace RGB  -profile "sRGB.icm"   -depth 8 -type TrueColor  -resize 2843x4000  -quality 100%   "preview.jpg"`

### System Configuration

- ImageMagick version: 7.0.10.16-portable-Q16
- Environment (Operating system, version and so on): Windows 10 Pro
- Additional information:

**Comment by urban-warrior** [comment]
The portable release of ImageMagick is designed to work on multiple chip sets and is not fully optimized for performance.  We compiled ImageMagick 7 on our Fedora host and your command completed in 13 seconds.

**Comment by VassiliYakovlev** [comment]
Thanks for your response. I found this https://www.imagemagick.org/discourse-server/viewtopic.php?t=28707
Is it probably the issue, that the portable version built without OpenMP?

**Comment by VassiliYakovlev** [comment]
I've build ImageMagick binaries with OpenMP, now it is faster - 26 sec, but still not so fast as installed. What I have to do in VisualStudio compiler related to build more optimized code?

**Comment by urban-warrior** [comment]
@dlemstra recommends adding `-define profile:highres-transform=false` to your command-line.  Does that help?

**Comment by VassiliYakovlev** [comment]
It doesn't help

**Comment by dlemstra** [comment]
Where did you add this? What is your complete command now?

**Comment by VassiliYakovlev** [comment]
I've added before all others:
`convert -define profile:highres-transform=false -density 150 -colorspace RGB "bigPDF.pdf[0]" -colorspace RGB -profile "sRGB.icm" -depth 8 -type TrueColor -resize 2843x4000 -quality 100% "preview.jpg"`

**Comment by VassiliYakovlev** [comment]
@dlemstra , is it possible to check-out the version 7.0.10.16?

**Comment by dlemstra** [comment]
Oh I did not notice that you want are using an older version. Can you upgrade to the latest version?

**Comment by VassiliYakovlev** [comment]
It looks like the portable version we downloaded ImageMagick-7.0.10-16-portable-Q16-x64 (non HDRI) was actually HDRI. Because of that there is confusing why the output different between the versions.

**Comment by VassiliYakovlev** [comment]
OK, I'l use the latest version.
How can I gain in my local build the same performance as installed ImageMagick-7.0.10-28-Q16-x64-static version? I've enabled OpenCL, OpenMP, disabled HDRI.

**Comment by urban-warrior** [comment]
Non-HDRI (integer) is generally faster than HDRI (floating point).  8-bit builds are generally faster than 16-bit.  Non-high precision profile transforms (integer) is generally faster than high precision (floating point).  Local builds are generally faster than builds we supply.  OpenMP-enabled builds are generally faster than single threaded versions of ImageMagick.  We lose some control to performance as we depend on Ghostscript to interpret the PDF language.  Some conversions directly from Ghostscript are faster than performing the conversion with ImageMagick.  And finally, some other image conversion software solutions are faster than ImageMagick because they are tuned for performance.  ImageMagick is tuned for accurate transforms, then performance.  See https://imagemagick.org/www/script/architecture.php.

**Comment by pietervp** [comment]
@urban-warrior Thanks for the excellent guidance! 
Ultimately we managed to get a build that has similar performance levels as the installed version. For reference, disabling OpenCL made a big impact for us, as we use IM primarily as a standalone exe. OpenCL was causing initialization costs on every command executed.

Still, we are seeing a performance 'regression' compared to our reference version 6.7.4.

Probably we are just facing progression in the library on the precision side, which impacts us at the performance side. 

For those interested, I attached a performance profiling snapshot of the convert command used in this example.
[Report20200903-0844.zip](https://github.com/ImageMagick/ImageMagick/files/5167554/Report20200903-0844.zip)
